src/prototype.rs

```diff
@@ -995,10 +1035,11 @@ fn parse_line<'a>(tree: &mut Tree<Item<'a>>, s: &'a str, mut ix: usize) -> (usiz
         match bytes[ix] {
             b'\n' | b'\r' => {
                 let mut i = ix;
-                if ix >= begin_text + 1 && bytes[ix - 1] == b'\\' {
+                let eol_bytes = scan_eol(&s[ix..]).0;
+                if ix >= begin_text + 1 && bytes[ix - 1] == b'\\' && ix + eol_bytes < s.len() {
                     i -= 1;
-                    tree.append_text(begin_text, i);
-                    ix += scan_eol(&s[ix..]).0;
+                    unescape_and_append(tree, s, begin_text, i);
```

Comment:
From an efficiency perspective, it seems like this ends up scanning plain text twice - once in this method, once in unescape. It's possible to do it once by including unescape triggers (`&`, probably add 0x00, the other two are `\` and `\r`, which are already covered) in the match.

But since this version is pretty clear and seems correct, I'm willing to take that up in a future performance pass.

In rough benchmarking, I'm seeing something like a 15% hit from this PR. As I say, acceptable for now, room for future improvement.

Reply:
Oof, I hadn't considered that. 15% is a pretty big hit. I guess we'll always have to make the trade-off between either scanning twice or allocating :thinking: Then the optimal strategy isn't as clear cut as I thought. Oh well, I learn!